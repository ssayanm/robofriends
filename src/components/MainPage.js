import React, { Component } from "react";

import CardList from "../components/CardList";
import SearchBox from "../components/SearchBox";
import Scroll from "../components/Scroll";
import ErrorBoundary from "../components/ErrorBoundary";
import "./MainPage.css";

class MainPage extends Component {
  componentDidMount() {
    this.props.onRequestRobots();
  }

  filterRobots = () => {
    return this.props.robots.filter(robot => {
      return robot.name
        .toLowerCase()
        .includes(this.props.searchField.toLowerCase());
    });
  };

  render() {
    const { onSearchChange, isPending } = this.props;

    return isPending ? (
      <h1>Loading</h1>
    ) : (
      <div className="App tc">
        <h1 className="f1">RoboFriends</h1>
        <SearchBox searchChange={onSearchChange} />
        <Scroll>
          <ErrorBoundary>
            <CardList robots={this.filterRobots(this.props.robots)} />
          </ErrorBoundary>
        </Scroll>
      </div>
    );
  }
}

export default MainPage;
