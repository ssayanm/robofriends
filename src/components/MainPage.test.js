import { shallow } from "enzyme";
import React from "react";
import MainPage from "./MainPage";

let wrapper;
beforeEach(() => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [],
    searchField: "a",
    isPending: false
  };
  wrapper = shallow(<MainPage {...mockProps} />);
});

it("expect to render MainPage component without crashing", () => {
  expect(wrapper.debug()).toMatchSnapshot();
});

it("filters robots correctly", () => {
  const mockProps2 = {
    onRequestRobots: jest.fn(),
    robots: [{ id: 2, name: "Jake", email: "jake@gmail.com" }],
    searchField: "Jake",
    isPending: false
  };
  const wrapper2 = shallow(<MainPage {...mockProps2} />);
  expect(wrapper2.instance().filterRobots([])).toEqual([
    {
      id: 2,
      name: "Jake",
      email: "jake@gmail.com"
    }
  ]);
});

it("filters robots correctly 2", () => {
  const mockProps3 = {
    onRequestRobots: jest.fn(),
    robots: [{ id: 2, name: "Jake", email: "jake@gmail.com" }],
    searchField: "o",
    isPending: false
  };
  const filteredRobots = [];
  const wrapper3 = shallow(<MainPage {...mockProps3} />);
  expect(wrapper3.instance().filterRobots([])).toEqual(filteredRobots);
});
