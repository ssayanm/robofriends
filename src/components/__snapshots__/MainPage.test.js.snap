// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render MainPage component without crashing 1`] = `
"<div className=\\"App tc\\">
  <h1 className=\\"f1\\">
    RoboFriends
  </h1>
  <SearchBox searchChange={[undefined]} />
  <Scroll>
    <ErrorBoundary>
      <CardList robots={{...}} />
    </ErrorBoundary>
  </Scroll>
</div>"
`;
